src/main/java/com/clouway/push/adapter/firebase/HttpFirebaseClient.java

```diff
@@ -0,0 +1,51 @@
+package com.clouway.push.adapter.firebase;
+
+import com.clouway.push.core.ChannelFailureException;
+import com.clouway.push.core.EventMessage;
+import com.clouway.push.core.HttpClient;
+import com.clouway.push.core.TokenGenerator;
+import com.google.api.client.http.ByteArrayContent;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpTransport;
+import com.google.gson.Gson;
+
+import java.io.IOException;
+
+public class HttpFirebaseClient implements HttpClient {
+  private String firebaseDbUrl;
+  private HttpTransport httpTransport;
+  private TokenGenerator tokenGenerator;
+
+  public HttpFirebaseClient(HttpTransport httpTransport, TokenGenerator tokenGenerator, String firebaseDbUrl) {
+    this.httpTransport = httpTransport;
+    this.tokenGenerator = tokenGenerator;
+    this.firebaseDbUrl = firebaseDbUrl;
+  }
+
+  public HttpResponse sendMessage(String namespace, String eventKey, EventMessage eventMessage) {
+    HttpRequestFactory requestFactory = httpTransport.createRequestFactory();
+    String accessToken = tokenGenerator.generateAccessToken();
+
+    GenericUrl url = new GenericUrl(
+            String.format("%s%s/%s.json?access_token=" + accessToken, firebaseDbUrl, namespace, eventKey));
+
+    HttpResponse response;
+
+    try {
+      String objectJson = new Gson().toJson(eventMessage);
```

Comment:
what is the reason aobut the objectJson - the  EventMessage contains only single field - message ?